### PR TITLE
ani-cli: only add the player to the PATH as a fallback, format, add diniamo as a maintainer

### PR DESCRIPTION
**Pull request by diniamo**
## Description of changes

- Removes the assertation for choosing at least one player. This is questionable, because this should only be done if the user can assure that a player is installed separately, which doesn't sound pure/correct. Should I undo this?
- Adds the player as a suffix in PATH, so the user's configuration takes precedence. This important, because for example in mpv, scripts are added through wrapping.
- Format with nixfmt-rfc-style
- Avoid using `with lib;`
- Avoid using `rec`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/ani-cli/default.nix [hunk 1]
```diff
@@ -1,63 +1,72 @@
...
-      ++ lib.optional chromecastSupport catt
-      ++ lib.optional syncSupport syncplay;
+  runtimeInputs = [
+    gnugrep
+    gnused
+    curl
+    fzf
+    ffmpeg
+    aria2
+  ] ++ lib.optional chromecastSupport catt ++ lib.optional syncSupport syncplay;
```
Comment (fufexan) [inline pkgs/applications/video/ani-cli/default.nix:47]:
```suggestion
  runtimeInputs = lib.flatten [
    gnugrep
    gnused
    curl
    fzf
    ffmpeg
    aria2
    (lib.optional chromecastSupport calt)
    (lib.optional syncSupport syncplay)
  ];
```

File: pkgs/applications/video/ani-cli/default.nix [hunk 2]
```diff
@@ -1,63 +1,72 @@
...
 
   installPhase = ''
     runHook preInstall
 
     install -Dm755 ani-cli $out/bin/ani-cli
 
     wrapProgram $out/bin/ani-cli \
-      --prefix PATH : ${lib.makeBinPath runtimeDependencies}
+      --prefix PATH : ${lib.makeBinPath finalAttrs.runtimeInputs} \
+      ${lib.optionalString (builtins.length players > 0) "--suffix PATH : ${lib.makeBinPath players}"}
```
Comment (fufexan) [inline pkgs/applications/video/ani-cli/default.nix:56]: Doesn't this break if all players are disabled due to the trailing `\`?

File: pkgs/applications/video/ani-cli/default.nix (hunk 1 above)
Comment (diniamo) [reply pkgs/applications/video/ani-cli/default.nix:47]: Same here, I don't like the added complexity.

File: pkgs/applications/video/ani-cli/default.nix [hunk 3]
```diff
@@ -1,63 +1,72 @@
...
+  withMpv ? true,
+  withVlc ? false,
+  withIina ? false,
+  chromecastSupport ? false,
+  syncSupport ? false,
 }:
 
-assert withMpv || withVlc || withIina;
+let
+  players = lib.optional withMpv mpv ++ lib.optional withVlc vlc ++ lib.optional withIina iina;
```
[the remark this answers is not in the document]
Comment (diniamo) [reply pkgs/applications/video/ani-cli/default.nix:25]: I don't like the added complexity.

**Review by skykanin — APPROVED** [review]

**Comment by diniamo** [comment]
Thanks

**Pull request merged**